kong/pdk/private/node.lua

```diff
@@ -0,0 +1,86 @@
+local utils = require "kong.tools.utils"
+local pl_file = require "pl.file"
+local pl_path = require "pl.path"
+local pl_dir = require "pl.dir"
+
+local ngx = ngx
+
+local cached_node_id
+
+local function node_id_filename(prefix)
+  return pl_path.join(prefix, "kong.id")
+end
+
+
+local function initialize_node_id(prefix)
+  if not pl_path.exists(prefix) then
+    local ok, err = pl_dir.makepath(prefix)
+    if not ok then
+      return false, "failed to create directory " .. prefix .. ": " .. err
+    end
+  end
+
+  local filename = node_id_filename(prefix)
+
+  if not pl_path.exists(filename) then
+    local id = utils.uuid()
+    ngx.log(ngx.INFO, "persisting node id " .. id .. " to filesystem ", filename)
+    local ok, write_err = pl_file.write(filename, id)
+    if not ok then
+      return false, "failed to persist node id to filesystem " .. filename .. ": " .. write_err
+    end
+    cached_node_id = id
+  end
+
+  return true, nil
```

Comment:
```suggestion
  return true
```

We don't usually write the `nil` for `err` explicitly.